.github/workflows/ci-docs-tests.yml

```diff
@@ -60,12 +60,25 @@ jobs:
         poetry-version: ${{ env.POETRY_VERSION }}
 
     - name: Load Poetry Cached Libraries ⬇
+      id: cache-poetry
       uses: actions/cache@v2
       with:
-        path: ~/.cache/pypoetry/virtualenvs
+        path: .venv
         key: ${{ runner.os }}-poetry-${{ env.POETRY_VERSION }}-${{ env.DEFAULT_PYTHON_VERSION }}-${{ hashFiles('**/poetry.lock') }}-${{ secrets.POETRY_CACHE_VERSION }}
         restore-keys: ${{ runner.os }}-poetry-${{ env.DEFAULT_PYTHON_VERSION }}
 
+    - name: Clear Poetry cache
+      if: steps.cache-poetry.outputs.cache-hit == 'true' && contains(github.event.pull_request.labels.*.name, 'tools:clear-poetry-cache-docs-tests')
+      run: rm -r .venv
+
+    - name: Create virtual environment
+      if: (steps.cache-poetry.outputs.cache-hit != 'true' || contains(github.event.pull_request.labels.*.name, 'tools:clear-poetry-cache-docs-tests'))
+      run: python -m venv create .venv
+
+    - name: Set up virtual environment
+      if: needs.changes.outputs.docs == 'true'
+      run: poetry config virtualenvs.in-project true
```

Comment:
Is it correct to set this `in-project` option to True? As far as I can see, [poetry will create a new virtualenv](https://python-poetry.org/docs/configuration/#virtualenvsin-project) itself with this option and it will be overwritten by `python -v venv create .venv` later on I'm guessing? If not set -which is the default- it will use existing venv if available (which will exist after manual creation and if it had not existed I think poetry would be creating a new venv by default and this is handled by `virtualenvs.create` option).

Reply:
I think this is bypassing the creation of the venv in poetry somehow, because this is what failed before
> If set to true, the virtualenv wil be created and expected in a folder named .venv within the root directory of the project.

In the case the `.venv` already exists (which we now enforce), I'm guessing poetry doesn't do anything (`expected in a folder named .venv`)